pkg/model/crd.go

```diff
@@ -371,6 +371,34 @@ func (r *CRD) ExceptionCode(httpStatusCode int) string {
 	return "UNKNOWN"
 }
 
+/**
+This method returns the required fields names for ReadOneInput which are present in ko.Status .
+During initial creation of aws resource, ReadOne call should be ignored if validation error
+occurs from absence of these fields.
+
+Example output: requiredStatusFields := []string{"GetApiInput.ApiId"}
+*/
+func (r *CRD) RequiredStatusFieldsForReadOneInput(indentLevel int) string {
+	op := r.Ops.ReadOne
+	inputShape := op.InputRef.Shape
+	if inputShape == nil || len(inputShape.Required) == 0 {
+		return ""
+	}
+	requiredFields := inputShape.Required
+	requiredStatusFieldsStr := ""
+	for _, requiredField := range requiredFields {
+		_, found := r.StatusFields[requiredField]
+		if found {
+			requiredStatusFieldsStr += fmt.Sprintf("\"%s.%s\",", inputShape.ShapeName, requiredField)
+		}
+	}
+	if last := len(requiredStatusFieldsStr) - 1; last >= 0 && requiredStatusFieldsStr[last] == ',' {
+		requiredStatusFieldsStr = requiredStatusFieldsStr[:last]
+	}
+	indent := strings.Repeat("\t", indentLevel)
+	return fmt.Sprintf("%srequiredStatusFields := []string{%s}", indent, requiredStatusFieldsStr)
+}
```

Comment:
I actually thought that what we were going to do was output Go code that would return NotFound immediately in `sdkFind()` if there were missing fields in the Input shape.

Currently, the top of the `sdkFind()` method looks like this:

```go
// sdkFind returns SDK-specific information about a supplied resource
func (rm *resourceManager) sdkFind(
	ctx context.Context,
	r *resource,
) (*resource, error) {
	input, err := rm.newDescribeRequestPayload(r)
	if err != nil {
		return nil, err
	}
...
```

what I thought we wanted to do was modify the code that generates the `newDescribeRequestPayload()` so that it returns `ackerr.NotFound` when certain fields that comprise required elements of the Input shape are missing.

```go

// newDescribeRequestPayload returns SDK-specific struct for the HTTP request
// payload of the Describe API call for the resource
func (rm *resourceManager) newDescribeRequestPayload(
	r *resource,
) (*svcsdk.DescribeAPIInput, error) {
	res := &svcsdk.DescribeAPIInput{}

        if r.ko.Status.APIID == nil {
                return ackerr.NotFound
        }

	return res, nil
}
```